compiler/rustc_codegen_cranelift/src/constant.rs

```diff
@@ -167,6 +167,7 @@ pub(crate) fn codegen_const_value<'tcx>(
     }
 
     match const_val {
+        ConstValue::ZST => unreachable!(), // we already handles ZST above
```

Comment:
Does `layout.is_zst()` always imply `ConstValue::ZST`? If so moving `CValue::by_ref(crate::Pointer::dangling(layout.align.pref), layout)` into this branch will be a tiny bit faster at runtime and reduce compile time of cg_clif itself a bit too.